operator/controllers/kyma_controller.go

```diff
@@ -137,29 +140,76 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, logger *logr
 	return nil
 }
 
-func (r *KymaReconciler) HandleDeletingState(_ context.Context) error {
-	return nil
-}
+func (r *KymaReconciler) HandleDeletingState(ctx context.Context, logger *logr.Logger, kyma *operatorv1alpha1.Kyma) (bool, error) {
+	templates, err := release.GetTemplates(r, ctx, kyma)
+	if err != nil {
+		return false, r.KymaStatus().UpdateStatus(ctx, kyma, operatorv1alpha1.KymaStateError,
+			"deletion cannot proceed - templates could not be fetched")
+	}
+
+	for _, component := range kyma.Spec.Components {
+		lookupResult := templates[component.Name]
+		if lookupResult == nil {
+			return false, fmt.Errorf("deletion cannot proceed - could not find template %s for resource %s",
+				component.Name, client.ObjectKeyFromObject(kyma))
+		}
 
-func (r *KymaReconciler) HandleErrorState(_ context.Context, logger *logr.Logger, kyma *operatorv1alpha1.Kyma) error {
-	if kyma.Status.ObservedGeneration == kyma.Generation {
-		logger.Info("skipping reconciliation for " + kyma.Name + ", already reconciled!")
+		desiredComponentStruct := &lookupResult.Template.Spec.Data
+		desiredComponentStruct.SetName(component.Name + "-name")
+		desiredComponentStruct.SetNamespace(kyma.GetNamespace())
+
+		actualComponentStruct := desiredComponentStruct.DeepCopy()
+
+		if err = r.Get(ctx, client.ObjectKeyFromObject(actualComponentStruct), actualComponentStruct); err == nil {
+			// component CR still exists
+			logger.Info(fmt.Sprintf("deletion cannot proceed - waiting for component CR %s to be deleted for %s",
```

Comment:
I think we should write this as a DeletionCondition into the status instead of just logging here, then we can immediately see in the Kyma why the deletions is still processing, so not only "ReadyCondition" but also "DeletionCondition". What do you think?